installer.lua

```diff
@@ -369,15 +369,34 @@ local function install()
                 return
             end
 
-            vim.notify("Installing rocks.nvim...")
-            local sc = vim.system({
+            local rocks_binaries_supported_arch_map = {
+                Darwin = {
+                    arm64 = "macosx-aarch64",
+                    aarch64 = "macosx-aarch64",
+                    x86_64 = "macosx-x86_64",
+                },
+                Linux = {
+                    x86_64 = "linux-x86_64",
+                },
+                Windows_NT = {
+                    x86_64 = "win32-x86_64",
+                },
+            }
+            local uname = vim.uv.os_uname()
+            local supported_arch = rocks_binaries_supported_arch_map[uname.sysname][uname.machine]
+
+            local install_cmd = {
                 luarocks_binary,
                 "--lua-version=5.1",
                 "--tree=" .. install_path,
-                "--server='https://nvim-neorocks.github.io/rocks-binaries/'",
                 "install",
                 "rocks.nvim",
-            }):wait()
+            }
+            if supported_arch then
+                table.insert(install_cmd, 4, "--server='https://nvim-neorocks.github.io/rocks-binaries/'")
+            end
+            vim.notify("Installing rocks.nvim...")
+            local sc = vim.system(install_cmd):wait()
```

Comment:
This logic is repeated and also present in `bootstrap.lua`
Might be worthwhile to extract this to a separate Lua module to keep things DRY

Reply:
Thanks for the suggestion. The bootstrap and install scripts are standalone and can't make use of module imports. That's why they have a lot of repeated logic.